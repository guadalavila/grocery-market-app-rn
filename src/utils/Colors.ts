export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
