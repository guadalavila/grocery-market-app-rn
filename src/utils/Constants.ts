export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
