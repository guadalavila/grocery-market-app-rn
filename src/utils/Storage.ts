import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
