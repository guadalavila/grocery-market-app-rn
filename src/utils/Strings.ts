export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
