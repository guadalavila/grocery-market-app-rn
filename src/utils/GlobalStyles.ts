import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
