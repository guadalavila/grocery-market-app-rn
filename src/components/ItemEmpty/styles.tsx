import React from 'react';
import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes, FontSize } from '../../utils/GlobalStyles';
export const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
    },
    text: {
        fontSize: FontSize.XXL,
        color: Colors.grey,
        marginHorizontal: Sizes.L,
        textAlign: 'center',
        fontWeight: 'bold',
    },
    containerIcon: {
        alignSelf: 'center',
        marginBottom: Sizes.XL,
    },
});
