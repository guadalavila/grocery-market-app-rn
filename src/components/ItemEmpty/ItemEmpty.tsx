import React from 'react';
import { Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Colors } from '../../utils/Colors';
import { styles } from './styles';

interface IItemEmptyProps {
    icon: string;
    message: string;
}

const ItemEmpty = ({ icon, message }: IItemEmptyProps) => {
    return (
        <View style={styles.container}>
            <View style={styles.containerIcon}>
                <Icon testID='icon' name={icon} size={130} color={Colors.greyLight} />
            </View>
            <Text style={styles.text}>{message}</Text>
        </View>
    );
};

export default ItemEmpty;
