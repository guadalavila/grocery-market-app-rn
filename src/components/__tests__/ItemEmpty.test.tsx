import { render, screen } from '@testing-library/react-native';
import React from 'react';
import { Colors } from '../../utils/Colors';
import ItemEmpty from '../ItemEmpty/ItemEmpty';

describe('test ItemEmpty component', () => {
    const icon = 'add';
    const message = 'Cart is empty!';

    test('should match snapshot', () => {
        const component = render(<ItemEmpty icon={icon} message={message} />);
        expect(component).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<ItemEmpty icon={icon} message={message} />);
        expect(screen).toBeTruthy();
    });

    test('should render title', () => {
        render(<ItemEmpty icon={icon} message={message} />);
        expect(screen.getByText(message)).toBeTruthy();
    });

    test('should render icon', () => {
        render(<ItemEmpty icon={icon} message={message} />);
        expect(screen.getByTestId('icon')).toBeTruthy();
        expect(screen.getByTestId('icon').props.name).toBe('add');
        expect(screen.getByTestId('icon').props.size).toBe(130);
        expect(screen.getByTestId('icon').props.color).toBe(Colors.greyLight);
    });
});
