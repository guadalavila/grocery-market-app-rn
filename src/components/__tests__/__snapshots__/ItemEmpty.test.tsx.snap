// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemEmpty component should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginBottom": 30,
      }
    }
  >
    <MockedIoniconsIcons
      color="#C5C5C5"
      name="add"
      size={130}
      testID="icon"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#9f9f9f",
        "fontSize": 26,
        "fontWeight": "bold",
        "marginHorizontal": 24,
        "textAlign": "center",
      }
    }
  >
    Cart is empty!
  </Text>
</View>
`;
